File: REAP-T/+MPCFunctions/initialize.m
function [Ad, Bd, Cd,Dd, NoS, NoI, NoO] = initialize(A, B, C, D, DeltaT)
    % Initialize and discretize system matrices
    G = ss(A, B, C, D);
    Gd = c2d(G, DeltaT);
    Ad = Gd.A;
    Bd = Gd.B;
    Cd = Gd.C;
    Dd=Gd.D;
    
    NoS = size(Ad, 1);
    NoI = size(Bd, 2);
    NoO = size(Cd, 1);

        Ad = [Ad Bd; zeros(NoI, NoS) zeros(NoI, NoI)];
    Bd = [zeros(NoS, NoI); eye(NoI)];
    Cd = [Cd Dd];

        NoS = size(Ad, 1);
    NoI = size(Bd, 2);
    NoO = size(Cd, 1);
end

File: REAP-T_Python/+MPCFunctions/initialize.py
import numpy as np
    
def initialize(A = None,B = None,C = None,D = None,DeltaT = None): 
    # Initialize and discretize system matrices
    G = ss(A,B,C,D)
    Gd = c2d(G,DeltaT)
    Ad = Gd.A
    Bd = Gd.B
    Cd = Gd.C
    Dd = Gd.D
    NoS = Ad.shape[1-1]
    NoI = Bd.shape[2-1]
    NoO = Cd.shape[1-1]
    Ad = np.array([[Ad,Bd],[np.zeros((NoI,NoS)),np.zeros((NoI,NoI))]])
    Bd = np.array([[np.zeros((NoS,NoI))],[np.eye(NoI)]])
    Cd = np.array([Cd,Dd])
    NoS = Ad.shape[1-1]
    NoI = Bd.shape[2-1]
    NoO = Cd.shape[1-1]
    return Ad,Bd,Cd,Dd,NoS,NoI,NoO
    
    return Ad,Bd,Cd,Dd,NoS,NoI,NoO